lib/src/dbs/options.rs

```diff
@@ -80,18 +80,24 @@ impl Options {
 		self.db.as_ref().unwrap()
 	}
 
-	/// Create a new Options object for a subquery
+	/// Create a new Options object for a function/subquery/future/etc.
 	pub fn dive(&self) -> Result<Options, Error> {
-		if self.dive < cnf::MAX_RECURSIVE_QUERIES {
+		self.dive_n(1)
+	}
+
+	/// Create a new Options object for an abnormally expensive function/subquery/future/etc.
+	pub fn dive_n(&self, n: usize) -> Result<Options, Error> {
```

Comment:
Instead of `dive_n`, shall we just change `dive` to accept an `n` and specify explicitly on every invocation of `dive` the `n`/`cost`?

Reply:
Sounds good! In testing, it seems like subqueries benefit from `dive(2)`.